reference-implementation/lib/readable-stream.js

```diff
@@ -554,10 +554,14 @@ function ReadableStreamTee(stream, cloneForBranch2) {
   });
 
   function pullAlgorithm() {
-    return ReadableStreamDefaultReaderRead(reader).then(result => {
-      if (closed === true) {
-        return;
-      }
+    if (pulling === true) {
```

Comment:
You could use `ReadableStreamGetNumReadRequests(stream) > 0` instead of having a separate `pulling` variable to keep track of. However, using a `pulling` variable makes it more obviously correct, so on balance I prefer it.

Nit: I think `reading` is a better name than `pulling`, since it really reflects the duration of the call to ReadableStreamDefaultReaderRead.

Reply:
Fine by me, I'll rename it to `reading`.